sw/host/ot_certs/src/cwt.rs

```diff
@@ -0,0 +1,354 @@
+            ConstantType::Integer{..} => 1 + 8,
+            ConstantType::String{value} => 1 + 8 + value.len(),
+        }
+    }
+}
+
+/// Declaration of a variable that can be filled into the template.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum StructureType {
+    Item(String),
+    Map(IndexMap<String, StructureType>),
+    Array(Vec<StructureType>),
+}
+
+impl StructureType {
+    fn get_max_size(&self, variables: &IndexMap<String, VariableType>, constants: &IndexMap<String, ConstantType>) -> Result<usize> {
+        let lookup = |var_name| {
+            if let Some(var) = variables.get(var_name) {
+                return Ok(var.get_max_size())
+            }
+
+            if let Some(var) = constants.get(var_name) {
+                return Ok(var.get_max_size())
+            }
+
+            return Err(anyhow!("Cannot find {} in variables or constants", var_name));
+        };
+
+        match self {
+            StructureType::Item(var_name) => lookup(var_name),
+            StructureType::Map(map) => {
+                let mut size: usize = 1 + 8;
+                for (k, v) in map.iter() {
+                    size += lookup(k)?;
+                    size += v.get_max_size(variables, constants)?;
+                }
+
+                Ok(size)
+            },
+            StructureType::Array(arr) => {
+                let mut size: usize = 1 + 8;
+                for v in arr.iter() {
+                    size += v.get_max_size(variables, constants)?;
+                }
+
+                Ok(size)
+            },
+        }
+    }
+}
+
+pub fn load_cwt_template(path: &PathBuf) -> Result<CwtTemplate> {
+    // Load template.
+    let template_content = fs::read_to_string(path)
+        .with_context(|| format!("Could not load the template file {}", path.display()))?;
+
+    CwtTemplate::from_hjson_str(&template_content)
+}
+
+impl CwtTemplate {
+    pub fn from_hjson_str(content: &str) -> Result<CwtTemplate> {
+        Ok(deser_hjson::from_str(content)?)
+    }
+}
+
+fn generate_c_var(template: &CwtTemplate, indent: &str) -> Result<String> {
+    let mut vars = String::new();
+    for (key, var) in template.variables.iter() {
+        match var {
+            VariableType::ByteArray{..} => {
+                vars += &indoc::formatdoc! { r#"
+                {indent}uint8_t *{key};
+                {indent}size_t {key}_size;
+                "#};
+            },
+            VariableType::Integer => {
+                vars += &indoc::formatdoc! { r#"
+                {indent}int32_t {key};
+                "#};
+            },
+            VariableType::String{..} => {
+                vars += &indoc::formatdoc! { r#"
+                {indent}char *{key};
+                {indent}size_t {key}_size;
+                "#};
+            },
+        }
+    }
+
+    Ok(vars)
+}
+
+fn generate_header(from_file: &str, template: &CwtTemplate) -> Result<String> {
```

Comment:
Maybe document a bit what this function outputs in the header, ie a struct for variables, and two functions for the TBS and cert.